### PR TITLE
Speaker Section + Small mobile tweaks

### PR DESCRIPTION
**Pull request by SwiftyAlex**
Code is separated pretty cleanly, so should be able to see commit-by-commit whats going on, and if we want to revert some of the work and pull it out, we can!

The models are placeholders, im sure we'll swap them out when it comes to the API. It may also be cool to make the socials more dynamic so it doesn't have to be twitter, it could be IG or something.

### Variables

I pulled out a bunch of colors that would be generally swapped in dark mode, into variables. It would now be fairly trivial to swap to dark mode, we'd just need to swap out some of the assets to have a light and dark 🌚.

### Grid 

Pretty close to design, and will hide the CFP when its finished, automagically 🧙🏻

There's a couple things to decide on:
- I struggled with the fancy border, but I'm not actually sure on its value anyway  🤔Should we keep it? if so, maybe someone wants to throw a quick pass at sorting out a nicer CSS solution!
- The link has an issue with legibility in some sizes due to the fancy background images, due to the way these float we can't really react to placements, so we could do with deciding if we throw a nav-bar esque background behind this, or just make it black.

| Size | Preview |
| - | - |
| iPhone SE | iphone-se |
| iPhone 8 | iphone-8 |
| iPhone 8 Plus | iphone-8-plus |
| iPad mini 7.9" | ipad-mini-7 9 |
| iPad mini 12.9" | ipad-12 9 |
| Laptop | laptop-ish |
| Desktop | desktop |

### Mobile

After I corrected the x-overflow of the page, I found a very weird side-effect that a lot of the layout was dependent on some implicit sizing. Due to the flaky nature of this, I went in and quickly did some small fixes to optimise for mobile, and make sure desktop looked the same.

( Note the image only looks different on desktop Tito because of the order of the content, it hasn't _actually_ moved )

#### Tito 
| Device | Before | After |
| - | - | - |
| Mobile | before-tito-mobile | after-tito-mobile |
| Desktop | before-tito-desktop|after-tito-desktop| 

#### Footer
| Device | Before | After |
| - | - | - |
| Mobile | before-iphone-footer | after-iphone-footer |
| Desktop | before-desktop-footer|after-desktop-footer|

### CODE REVIEW

**Review by adamrushy — APPROVED** [review]
Looks great to me 🔥

**Comment by adamrushy** [comment]
2 comments actually:

1. Shall we hide the whole block of speakers when it's empty?
2. Shall we insert an empty array of speakers for now and include the CFP expiry date? That way we can deploy the website with these changes right now it throws an error :]

**Comment by SwiftyAlex** [comment]
> 2 comments actually:
> 
> 1. Shall we hide the whole block of speakers when it's empty?
> 2. Shall we insert an empty array of speakers for now and include the CFP expiry date? That way we can deploy the website with these changes right now it throws an error :]

One is actually already handled, it collapses the grid and just shows the meet the speakers section - do you mean a different kind of fallback, like a different section, or just this? 
Screenshot 2022-03-20 at 22 05 40

If you set the routes to this ( which is an empty array ) it'll achieve both of what you said above I believe @adamrushy 

```
func routes(_ app: Application) throws {
    app.get { req -> EventLoopFuture<View> in
        let cfpExpirationDate = Date(timeIntervalSince1970: 1651356000) // 30th April 22
        return req.view.render("Home/home", HomeContext(cfpEnabled: cfpExpirationDate > Date()))
    }
}
```

File: Public/css/style.css [hunk 1]
```diff
@@ -277,3 +295,177 @@ ul.tickets li a {
   text-transform: uppercase;
   font-size: 0.7em;
 }
+
+/* Speaker Grid */
+.speaker-grid-container {
+  width: 100%;
+  border: 1px solid var(--border-color);
+  padding: 80px 0px 80px 0px;
```
Comment (pdamonkey) [inline Public/css/style.css:303]:
```suggestion
  padding: 80px 0;
```

File: Resources/Views/Home/_speakers.leaf [hunk 2]
```diff
@@ -1,16 +1,44 @@
...
+          <a id="speakers">
+            <h2>
+              Meet the<br/>
+              speakers
+            </h2>
+          </a>
+          #if(cfpEnabled):
+          <div class="call-for-speakers">
+            <img src="/img/arrowicon.svg"/>
+            <a href="https://sessionize.com/swiftleeds/">Want to be here? get in touch.</a>
```
Comment (pdamonkey) [inline Resources/Views/Home/_speakers.leaf:14]: Should this be target blank to open up in a new tab?

File: Resources/Views/Home/_speakers.leaf [hunk 3]
```diff
@@ -1,16 +1,44 @@
...
+                        </div>
+                    </a>
+                    <p class="speaker-grid-name">#(speaker.name)</p>
+                    <p class="speaker-grid-title">#(speaker.title)</p>
+                </div>
+            </div>
+        #endfor
+     </div>
+    </div>
+    #endif
```
Comment (pdamonkey) [inline Resources/Views/Home/_speakers.leaf:42]: Should we have a backup in an else saying Awaiting Speakers or something similar?

File: Resources/Views/Home/home.leaf [hunk 4]
```diff
@@ -36,6 +36,8 @@
     <div class="container">
       #extend("Home/_about-us")
       <br/>
+      #extend("Home/_speakers")
```
Comment (pdamonkey) [inline Resources/Views/Home/home.leaf:39]: As we don't have the speakers yet, but we are about to push the tickets, this should probably go after the order tickets section.

**Comment by adamrushy** [comment]
> > 2 comments actually:
> > 
> > 1. Shall we hide the whole block of speakers when it's empty?
> > 2. Shall we insert an empty array of speakers for now and include the CFP expiry date? That way we can deploy the website with these changes right now it throws an error :]
> 
> One is actually already handled, it collapses the grid and just shows the meet the speakers section - do you mean a different kind of fallback, like a different section, or just this?
> 
> Edit: A thought on this, could be cool to have a separate section on the soon page which is just "Want to speak at SwiftLeeds? get in touch.", we could throw that together pretty easily
> 
> Screenshot 2022-03-20 at 22 05 40
> 
> If you set the routes to this ( which is an empty array ) it'll achieve both of what you said above I believe @adamrushy
> 
> ```
> func routes(_ app: Application) throws {
>     app.get { req -> EventLoopFuture<View> in
>         let cfpExpirationDate = Date(timeIntervalSince1970: 1651356000) // 30th April 22
>         return req.view.render("Home/home", HomeContext(cfpEnabled: cfpExpirationDate > Date()))
>     }
> }
> ```

Yeah, so right now I am getting an error unless I make the above changes, so I think we should return an empty array? You're right about the empty stack, I missed that part 👍🏼

**Comment by SwiftyAlex** [comment]
> Yeah, so right now I am getting an error unless I make the above changes, so I think we should return an empty array? You're right about the empty stack, I missed that part 👍🏼

We can only make that change ( as in, adding the content when we render the route ) when we switch the site to go live, as returning it from the route and passing the content would do that - do you want me to make that change here and we can just hold this PR from going in until we want to go live on the site? You'll get the empty block with the CTA 😄

Also apologies @adamrushy , when I tried to reply to your comment I somehow ended up editing it, so I've reset it to how it was 🙏🏻

File: Resources/Views/Home/_speakers.leaf (hunk 3 above)
Comment (SwiftyAlex) [reply Resources/Views/Home/_speakers.leaf:42]: It just empties the block, shows the title and the CTA. I think that should do it, but open to suggestions if we can think of something else? I personally think it gets the message across as-is but there's scope to improve for sure

File: Resources/Views/Home/_speakers.leaf (hunk 2 above)
Comment (SwiftyAlex) [reply Resources/Views/Home/_speakers.leaf:14]: It certainly can be!

**Review by jw1540 — APPROVED** [review]

**Pull request merged**